src/Sulu/Bundle/PreviewBundle/Resources/js/views/Preview/Preview.js

```diff
@@ -0,0 +1,265 @@
+import {sidebarStore} from 'sulu-admin-bundle/containers/Sidebar';
+import {FormStore} from 'sulu-admin-bundle/containers/Form';
+import ResourceStore from 'sulu-admin-bundle/stores/ResourceStore';
+import Requester from 'sulu-admin-bundle/services/Requester';
+import Toolbar from 'sulu-admin-bundle/components/Toolbar';
+import Loader from 'sulu-admin-bundle/components/Loader';
+import previewStyles from './preview.scss';
+import previewConfigStore from './stores/PreviewConfigStore';
+
+type Props = {
+    resourceStore: ResourceStore,
+    formStore: FormStore,
+    router: Router,
+};
+
+@observer
+export default class Preview extends React.Component<Props> {
+    @observable iframeRef: ?Object;
+    @observable token: ?string;
+    @observable started: boolean = false;
+
+    typeDisposer: () => void;
+    dataDisposer: () => void;
+
+    componentDidMount() {
+        if (previewConfigStore.mode === 'auto') {
+            this.startPreview();
+        }
+    }
+
+    @action startPreview = () => {
+        const {
+            resourceStore,
+            formStore,
+            router: {
+                attributes: {
+                    locale,
+                },
+            },
+        } = this.props;
+
+        const route = previewConfigStore.generateRoute('start', {
+            provider: resourceStore.resourceKey,
+            locale: locale,
+            id: resourceStore.id,
+        });
+        if (!route) {
+            return;
+        }
+
+        Requester.get(route).then((response) => {
+            this.setToken(response.token);
+        });
+
+        this.typeDisposer = autorun(() => {
+            if (formStore.loading || !this.iframeRef) {
+                return;
+            }
+
+            // $FlowFixMe
+            this.updateContext(formStore.type.get());
+        });
+
+        this.dataDisposer = autorun(() => {
+            if (resourceStore.loading || !this.iframeRef) {
+                return;
+            }
+
+            this.updatePreview(toJS(resourceStore.data));
+        });
+
+        this.started = true;
+    };
+
+    updatePreview = debounce((data) => {
+        const {
+            router: {
+                attributes: {
+                    locale,
+                    webspace,
+                },
+            },
+        } = this.props;
+
+        const route = previewConfigStore.generateRoute('update', {
+            locale: locale,
+            webspace: webspace,
+            token: this.token,
+        });
+        if (!route) {
+            return;
+        }
+
+        Requester.post(route, {data: data}).then((response) => {
+            const document = this.getPreviewDocument();
+            if (!document) {
+                return;
+            }
+
+            document.open();
+            document.write(response.content);
+            document.close();
+        });
+    }, previewConfigStore.debounceDelay);
+
+    updateContext = debounce((type) => {
```

Comment:
Does the `debounce` make sense here? That's nothing that happens so often, or is it?